Core/include/Acts/EventData/TrackStateType.hpp

```diff
@@ -39,18 +39,10 @@ class TrackStateType {
   using raw_type = std::uint64_t;
   static constexpr std::size_t kRawBits =
       std::numeric_limits<std::make_unsigned<raw_type>::type>::digits;
+
   /// Constructor from a reference to the underlying value container
   /// @param raw the value container
-  TrackStateType(raw_type& raw) : m_raw{&raw} {}
-
-  /// Assign the value from another set of flags
-  /// @param other the other set of flags to assign
-  /// @return this object
-  TrackStateType& operator=(const TrackStateType& other) {
-    assert(other.m_raw != nullptr);
-    *m_raw = *other.m_raw;
-    return *this;
-  }
```

Comment:
If this is removed, it might be safer to `delete` it. The default assignment operator will copy the pointer, which may not be what we want. There is an assignment operator from `ConstTrackStateType`, but I don't think that suppresses the default assignment.

Unless copying pointers is what is wanted, it is probably safer to implement or `delete` all the copy constructor and assignment operator for `TrackStateType` and `ConstTrackStateType`.

If copying pointers *is* what is wanted, then can we be explicit and `default` them all?

Reply:
I try to move these changes into another PR